src/compile/scale/domain.ts

```diff
@@ -208,7 +208,7 @@ function parseSingleChannelDomain(scaleType: ScaleType, domain: Domain, model: U
         // Use range if we added it and the scale does not support computing a range as a signal.
         field: model.field(channel, binRequiresRange(fieldDef, channel) ? {binSuffix: 'range'} : {}),
         // we have to use a sort object if sort = true to make the sort correct by bin start
-        sort: sort === true || !isSortField(sort) ? {
+        sort: sort !== false && (sort === true || !isSortField(sort)) ? {
```

Comment:
This is weird.  Shouldn't sort be true by default? 
If so, shouldn't this be `sort !== false && !isSortField(sort)`?

Reply:
Actually if sort is true by default and we should allow using null and false (just in case of user error), I guess this should be `sort !== null && sort !== false && !isSortField(sort)`?

Reply:
The first comment doesn't work because in `sort !== false && !isSortField(sort)` the second `sort` could be `true` and that doesn't pass type checking. 

Also no to the second comment! Because `sort` comes from `domainSort` and is thus already cleaned up.